### ISSUE TITLE
Option 'strictPropertyInitialization' cannot be specified without specifying option 'strictNullChecks'

### ISSUE THREAD

**Issue by marcomura** [Microsoft/TypeScript#24716]
Upgrading from 2.8.1 to 2.9.1, I'm now getting the error in the title.
Why strictPropertyInitialization now can't be specified without strictNullChecks? Is it a bug?

**Comment by a-tarasyuk** [comment]
@marcomura I think, it is not a bug, see #23659

**Comment by RyanCavanaugh** [comment]
SPI does nothing if SNC isn't on, so this warning is to tell you that you're not doing what you think you are

**Comment by marcomura** [comment]
Why strictPropertyInitialization should require stringNullChecks?
I don't see how the first should require the second...

```ts
class C {
    foo: number;
    bar = "hello";
    baz: boolean;
    //  ~~~
    //  Error! Property 'baz' has no initializer and is not assigned directly in the constructor.
    constructor() {
        this.foo = 42;
    }
}
```

Without stringNullChecks, I can solve the error explicitly assigning undefined to baz.

**Comment by marcomura** [comment]
Ping on this :)

**Comment by mhegazy** [comment]
Please see what `--strictNullChecks` really mean at https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript

**Comment by marcomura** [comment]
That's why I can't see its relation with --strictPropertyInitialization
